src/wasm/tree_parse.cpp

```diff
@@ -65,3 +66,46 @@ TreeNode prune_tree(const TreeNode &node, uint64_t threshold_left,
 
   return pruned_node;
 }
+
+
+
+TreeNode merge_nodes(TreeNode& node) {
+    std::unordered_map<std::string, TreeNode> grouped_children;
+
+    for (auto& child : node.children) {
+        TreeNode merged_child = merge_nodes(child);
+
+        if (grouped_children.find(merged_child.name) != grouped_children.end()) {
+            grouped_children[merged_child.name].value += merged_child.value;
+
+            grouped_children[merged_child.name].children.insert(
+                grouped_children[merged_child.name].children.end(),
+                merged_child.children.begin(),
+                merged_child.children.end()
+            );
+        } else {
+            grouped_children[merged_child.name] = merged_child;
+        }
+    }
```

Comment:
I think this code will produce a not-fully-merged flame graph for this time-ordered slice with arbitrary values:
![image](https://github.com/user-attachments/assets/bd239f91-56e8-4304-8dad-3693185ea565)

Could you check it and fix the code if the above turns out to be the case?

Reply:
Yes, sorry about that, should've started merging from root first. I hope it is ok now.